Make heartbeat message self-descriptive

luckysori opened the pull request:
Closes #1116.

Actually has nothing to do with losing connection to the maker specifically. My theory is that the change in the maker status means that delivering the next heartbeat is delayed by a little bit.

luckysori commented on taker-frontend/src/useEventSource.ts:
```diff
@@ -18,14 +18,14 @@ export function useEventSource(url: string): [EventSource | null, boolean] {
             timer = setTimeout(() => {
                 setIsConnected(false);
-            }, 5000);
+            }, 10000);
```
🤔 We could add a `next_heartbeat` field to the heartbeat message and use that value + a buffer here. That way this value can never go out of sync with the backend.

thomaseizinger replied:
Nit: `next` would imply an absolute value? Just `interval` (or `pulse`? 😁) perhaps, otherwise concept ACK.

luckysori committed:
Make SSE heartbeat self-descriptive

It now includes an `interval` field which the UI can use to decide if
the backend is missing.

luckysori committed:
Close EventSource before window reload

Otherwise we emit an annoying warning after reloading. The fix comes
from here: https://bugzilla.mozilla.org/show_bug.cgi?id=833462#c11.
